satellite/satellitedb/overlaycache.go

```diff
@@ -169,6 +169,43 @@ func (cache *overlaycache) GetAll(ctx context.Context, ids storj.NodeIDList) ([]
 	return infos, nil
 }
 
+// ReliableAndOnline filters a set of nodes to reliable and online nodes, independent of new
+func (cache *overlaycache) ReliableAndOnline(ctx context.Context, c *overlay.NodeCriteria, nodeIds storj.NodeIDList) (goodNodes map[storj.NodeID]bool, err error) {
```

Comment:
I'm not quite sure why you flipped the not? We expect most of the nodes from 100 to be online, so by getting `unreliable` we are transferring less data from database.

Reply:
Also return a list rather than a map. Checking against a short list might be faster on average rather than a map and involve less allocations.

Reply:
Instead of `c` use `criteria`

Reply:
I don't think that `crit` is better than `c`.
We have to eventually materialize some conventions for var names.

Reply:
https://github.com/storj/storj/wiki/Style#variable-naming

Reply:
The reason I mentioned it here because while I was reading:

```
args = append(args, c.AuditSuccessRatio, c.UptimeSuccessRatio, time.Now().Add(-overlay.OnlineWindow))
```

it wasn't immediately clear what `c.` was.

Reply:
I read those, but they are still unclear, considering just only one of the suggestions `crit`.

Materialize I meant to tune them.

Reply:
`cirt` sounds to me `critical`

Reply:
I would go with `criteria`.

Reply:
"I'm not quite sure why you flipped the not?"  Ugh.  In part, I was trying to make those test pass and apparently I stopped thinking.  The test include items which are not in the overlay cache at all... which you can't query for, so I reversed it.  Counting the `ReliableAndOnline()` nodes could also replace `.GetAll()`.

However, I can't say it was ever my intention to write the fastest code, it is to write less fragile code, which almost always means less code.  I genuinely believed that Fadila's GetAll() w/ Go logic solves this problem well.  I've seen query planners make terrible decisions with IN clauses and other criteria.  Perhaps the best answer is to leave things where they were.

Reply:
Fixed: Created ReliableAndOnline and UnreliableOrOffline SQL methods

Reply:
Renamed UnreliableOrOffline to KnownUnreliableOrOffline for clarity